src/UI/Art/SpriteArtist/PlayerSetupCommanderArtist.java

```diff
@@ -44,32 +45,37 @@ public static void draw(Graphics g, IController controller, ArrayList<CommanderI
     myG.drawImage(tooltip, myWidth - tooltip.getWidth(), 3, null);
 
     /////////////// Commander Portrait //////////////////////
-    int highlightedCommander = control.getSelectedCommander();
-    BufferedImage likeness = SpriteLibrary.getCommanderSprites( infos.get(highlightedCommander).name ).body;
+    int highlightedCmdr = control.cmdrSelector.getSelectionNormalized();
+    int highlightedCmdrOffset = control.cmdrSelector.getSelectionAbsolute();
+    BufferedImage likeness = SpriteLibrary.getCommanderSprites( infos.get(highlightedCmdr).name ).body;
     myG.drawImage(likeness, myWidth-likeness.getWidth(), myHeight-likeness.getHeight(), null);
 
     /////////////// Commander Panels //////////////////////
     // Calculate the vertical space each player panel will consume.
     int panelBuffer = 3;
     int panelHeight = CommanderPanel.PANEL_HEIGHT+panelBuffer;
 
-    // Find where the zeroth Commander should be drawn. Start by assuming it's centered.
-    int drawYCenter = myHeight / 2;
+    // Find where the zeroth Commander should be drawn.
+    panelOffsetY.set(highlightedCmdrOffset*panelHeight);
+    int drawYCenter = myHeight / 2 - panelOffsetY.geti();
 
     // We're gonna make this an endless scroll, so back up (in y-space and in the CO list) until
     // we find the Commander that would be off the top of the screen.
     OptionSelector coToDraw = new OptionSelector(infos.size());
-    coToDraw.setSelectedOption(highlightedCommander);
     while( drawYCenter + CommanderPanel.PANEL_HEIGHT/2 > 0 )
     {
       coToDraw.prev();
       drawYCenter -= panelHeight;
     }
-    // We don't actually want to draw something off the screen, so go forward one again.
-    coToDraw.next();
-    drawYCenter += panelHeight;
+    // We don't actually want to draw something off the screen, so go forward until we are on-screen again.
+    while( drawYCenter + CommanderPanel.PANEL_HEIGHT/2 < 0 )
+    {
+      coToDraw.next();
+      drawYCenter += panelHeight;
+    }
```

Comment:
Do these *both* need to be loops?

Reply:
Yes, because I'm using the CO selector's absolute position, which means my original offset could be very negative.

I can't use the normalized position because then the whole list snaps back in the other direction when you go past the end instead of scrolling forever.